### DISCUSSION TITLE
Side Navigation

### DISCUSSION THREAD

**Discussion by jeffreylauwers** [nl-design-system/backlog#242] [Component suggestions]
<img width="480" alt="visuele weergave van het side navigation component" src="https://github.com/nl-design-system/backlog/assets/545207/1773ed83-6367-4117-914f-9c3c017c3090">

We willen graag meer leren over het Side Navigation component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1713340674733069)

---

## Naam
Side Navigation

## Doel
Verticale navigatie die de mogelijkheid biedt om tussen pagina's te navigeren.

## Gerelateerde componenten
[Navigation Bar](https://github.com/orgs/nl-design-system/discussions/267)

## Onderzoek

[Left-Side Vertical Navigation on Desktop: Scalable, Responsive, and Easy to Scan - Nielsen Norman Group](https://www.nngroup.com/articles/vertical-nav/)
Artikel waarin het Side Navigation component wordt toegelicht aan de hand van voordelen en richtlijnen. Deze informatie komt ook terug in een [bijbehorende video](https://www.nngroup.com/videos/vertical-navigation/).

[UI shell left panel](https://carbondesignsystem.com/components/UI-shell-header/usage/)
Dit component is onderdeel van de de UI shell header. De header en side navigation zijn hierdoor onlosmakelijk aan elkaar verbonden.

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-sidenavigation--docs)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/SideNavigation)
- [Figma - Huidig](https://www.figma.com/file/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?type=design&node-id=1628-387&mode=design&t=5v1D11FMYhUfXZo3-4)
- [Figma - WIP](https://www.figma.com/file/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?type=design&node-id=1670-10259&mode=design&t=dByatFdC8tRtiv3u-4)

---

### Screenshots / Varianten

Storybook
<img width="1023" alt="Scherm­afbeelding 2024-04-22 om 10 41 33" src="https://github.com/nl-design-system/backlog/assets/545207/5cab5b9c-ff41-49a7-8b54-b9b152801e7b">

**Figma - Huidig**
States
<img width="1003" alt="Scherm­afbeelding 2024-04-22 om 10 42 11" src="https://github.com/nl-design-system/backlog/assets/545207/19006daf-d7d6-4938-93e3-52d47dc65c97">

**Figma - WIP**
Default
<img width="978" alt="Scherm­afbeelding 2024-04-22 om 12 14 51" src="https://github.com/nl-design-system/backlog/assets/545207/5f13c5f5-2c20-457b-9aec-fb38d0c8de4a">

Groeperen
<img width="971" alt="Scherm­afbeelding 2024-04-22 om 12 14 58" src="https://github.com/nl-design-system/backlog/assets/545207/32e583e4-7b7b-48d4-9adf-80331591716e">

States (default)
<img width="498" alt="Scherm­afbeelding 2024-04-22 om 12 15 25" src="https://github.com/nl-design-system/backlog/assets/545207/e130a08a-ae72-4e3e-9fdb-0883c9ace3a1">

States (current)
<img width="497" alt="Scherm­afbeelding 2024-04-22 om 12 15 31" src="https://github.com/nl-design-system/backlog/assets/545207/68ac8413-3ff5-409f-adfc-2453809d9c9b">

---

### Notities
- Per item optie voor icoon, couter badge
- In Figma - Huidig ook de mogelijkheid voor een chervron down om aan te geven dat er sub items zijn.
- Items hebben hebben bij default state een donkergrijze kleur.
- Documentatie beschikbaar.

**Reply by Rozerinay** [reply]
## Toelichting op het gebruik van Side navigation bij Den Haag

Op basis van een uitwisseling in de NL Design System Figma bibliotheek:

- De Side Navigation wordt bij gemeente Den Haag alleen gebruikt in MijnDenHaag en is bedoeld voor navigatie tussen pagina’s.
- De Anchor Navigation, zoals te zien op [denhaag.nl](https://www.denhaag.nl/nl/belastingen/aanmaning-belastingen/#aanmaning-betalen), wordt gebruikt voor navigatie binnen dezelfde pagina. Deze component toont de koppen op de pagina en werkt met ankerlinks.

Bron: uitwisseling tussen @Rozerinay en Marieke in de NL Design System Figma bibliotheek (2024).

**Reply by Rozerinay** [reply]
Opmerking afkomstig uit een comment in de NL Design System Figma bibliotheek:

Token `denhaag.sidenav.link.current.color` is niet opgenomen in het voorbeeldthema (JSON) in Figma, omdat de CSS-variabelenaam voor ‘current’ gelijk blijkt aan ‘active’. Beide tokens zijn aangemaakt.

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/navigation)

---

### Screenshots / Varianten

**Documentatie website**
Zijnavigatie
Een navigatiebalk links van de content. Subnavigatie is alleen bedoeld voor in-page navigatie.
<img width="922" alt="Scherm­afbeelding 2024-04-22 om 12 55 50" src="https://github.com/nl-design-system/backlog/assets/545207/10a9618d-c040-4343-969b-383871980488">

---

### Notities
- Als 'Variant' binnen het Navigation component onder de naam 'Zijnavigatie'.
- Vanwege beschrijving lijkt deze variant enkel bedoel voor 'anchor' navigatie.
- Optie voor sub-items
- Geen 'current' weergave waar ook duidt op 'anchor' navigatie.

**Comment by jeffreylauwers** [comment]
## Rotterdam

- [Storybook](https://nl-design-system.github.io/rotterdam/?path=/docs/css-denhaag-sidenav--docs)
- [GitHub](https://github.com/nl-design-system/rotterdam/tree/main/rotterdam-nlds-parent-wicket/docs-wicket/src/main/java/nl/rotterdam/nl_design_system/docs/wicket/side_nav)
- [Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=19266-13825&mode=design&t=XZtnm5lDHJQ0gdqj-4)

---

### Screenshots / Varianten

**Storybook**

<img width="1024" alt="Scherm­afbeelding 2024-04-22 om 13 10 56" src="https://github.com/nl-design-system/backlog/assets/545207/f497e3a4-1fea-41ab-b080-7acaa83f61f4">


**Figma**

Default
<img width="766" alt="Scherm­afbeelding 2024-04-22 om 13 11 54" src="https://github.com/nl-design-system/backlog/assets/545207/c310866b-a7eb-4ca9-ace7-2d1dba0a130b">

With icons
<img width="587" alt="Scherm­afbeelding 2024-04-22 om 13 12 01" src="https://github.com/nl-design-system/backlog/assets/545207/715ff204-e8ad-43df-9750-edaf6e57d29c">

States
<img width="974" alt="Scherm­afbeelding 2024-04-22 om 13 16 16" src="https://github.com/nl-design-system/backlog/assets/545207/1449132b-722d-45b0-9bb1-72aac34b364b">


---

### Notities
- Storybook op basis van Den Haag sidenavigation.
- Figma component wijkt af van Storybook. Zonder icoon bevat border inline-start. Met icoon dan geen border.

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-sidenav--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/navigatie-sidenav)
- [Figma](https://www.figma.com/file/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?type=design&node-id=153-1140&mode=design&t=QPCkq0xOt8SFquC4-4)

---

### Screenshots / Varianten

**Storybook**

Default
<img width="1023" alt="Scherm­afbeelding 2024-04-22 om 13 21 27" src="https://github.com/nl-design-system/backlog/assets/545207/601d2f35-1ae8-4fad-8d84-518162962080">

Children
<img width="1023" alt="Scherm­afbeelding 2024-04-22 om 13 21 33" src="https://github.com/nl-design-system/backlog/assets/545207/895efc12-08df-4b53-89f3-11a3303f5a0b">

Siblings
<img width="1028" alt="Scherm­afbeelding 2024-04-22 om 13 21 39" src="https://github.com/nl-design-system/backlog/assets/545207/fd94b46c-c2c7-46a1-9999-10b5930f617e">

Children and siblings
<img width="1028" alt="Scherm­afbeelding 2024-04-22 om 13 21 46" src="https://github.com/nl-design-system/backlog/assets/545207/af2f9a44-230f-4a20-9bdc-a3576fdf6c9a">

**Figma**

Default
<img width="981" alt="Scherm­afbeelding 2024-04-22 om 13 23 27" src="https://github.com/nl-design-system/backlog/assets/545207/274f6a09-50fb-48b6-9add-9dff8c35483c">

Groeperen
<img width="976" alt="Scherm­afbeelding 2024-04-22 om 13 23 36" src="https://github.com/nl-design-system/backlog/assets/545207/8207e5d6-49fc-4485-9f8d-d12ae21627ac">

States (default)
<img width="501" alt="Scherm­afbeelding 2024-04-22 om 13 23 47" src="https://github.com/nl-design-system/backlog/assets/545207/b07d87f6-06c7-48e3-9353-04ff5ef2414f">

States (current)
<img width="495" alt="Scherm­afbeelding 2024-04-22 om 13 23 54" src="https://github.com/nl-design-system/backlog/assets/545207/977ab8c5-1296-450c-8253-c616cc3333ff">

---

### Notities
- Documentatie beschikbaar.
- Figma component op basis van Den Haag.

**Comment by jeffreylauwers** [comment]
## DSO Toolkit

Niet als component. Wel op website zelf. Bijvoorbeeld:

- https://aandeslagmetdeomgevingswet.nl/ontwikkelaarsportaal/dso/dso/digitaal-stelsel/
- https://aandeslagmetdeomgevingswet.nl/ontwikkelaarsportaal/presentaties-documenten/leveranciersdag-26-maart-2024/

---

### Screenshots / Varianten

**Website**

Default
<img width="639" alt="Scherm­afbeelding 2024-04-22 om 13 29 14" src="https://github.com/nl-design-system/backlog/assets/545207/7e4311a5-d0b3-44dc-98bd-206b821ef3e1">

Hover
<img width="706" alt="Scherm­afbeelding 2024-04-22 om 13 29 20" src="https://github.com/nl-design-system/backlog/assets/545207/218f0d28-cb8f-4e22-950e-53035c4117a6">

Focus
<img width="770" alt="Scherm­afbeelding 2024-04-22 om 13 30 10" src="https://github.com/nl-design-system/backlog/assets/545207/3de4d5b6-82f9-41b4-af6b-cfa979304da2">

Sub items
<img width="657" alt="Scherm­afbeelding 2024-04-22 om 13 31 15" src="https://github.com/nl-design-system/backlog/assets/545207/b52c8a6a-bbc7-403b-8f42-bac5c71889dd">

**Comment by Rozerinay** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Side Navigation op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=side+navigation)

**Comment by emma-koole** [comment]
## Carbon
[Storybook](https://web-components.carbondesignsystem.com/?path=/story/components-ui-shell--header-base-w-navigation-actions-and-side-nav)

---

### Screenshots / Varianten

**Storybook**

Header Base with Navigation, Actions and SideNav
![Header Base w: Navigation, Actions and SideNav](https://github.com/user-attachments/assets/04aefe48-b967-46d0-9871-264b554ccf6b)

Header Base with Navigation, Actions and SideNav - menu item active
![Header Base w: Navigation, Actions and SideNav - menu item active](https://github.com/user-attachments/assets/567793b9-60f1-4378-91c2-de1172552310)

Side Nav Rail
![SideNav Rail](https://github.com/user-attachments/assets/11c1dff5-0987-4be6-abd8-92d6dd55f378)

Side Nav Rail Hover
![SideNav Rail - hover](https://github.com/user-attachments/assets/eaf1d2c3-b66d-4b71-9f9d-282c621b6c11)

SideNav Rail with Header
![SideNav Rail w: Header](https://github.com/user-attachments/assets/5bda6607-6ff8-49e5-abbb-387ee28e24b2)

SideNav Rail with Header - hover
![SideNav Rail w: Header - hover](https://github.com/user-attachments/assets/ef4c1723-2da7-4a37-b654-60cb417f76d3)

Fixed SideNav with Divider
![Fixed SideNav w:Divider ](https://github.com/user-attachments/assets/091d684a-e2ca-480e-90e8-fced911a76e6)

SideNav w/ large side nav items
![SideNav w: large side nav items](https://github.com/user-attachments/assets/f398f5b1-4d2b-4185-a3eb-d297c6fc05b1)

SideNav w/ large side nav items - menu list item active
![SideNav w: large side nav items - list item active](https://github.com/user-attachments/assets/076cc95c-2cad-449c-a7f4-e1c40232a5e1)

SideNav w/ large side nav items - hover on list item
![SideNav w: large side nav items - hover on list item](https://github.com/user-attachments/assets/f294be1c-dc39-4ad0-8452-149803959479)

## Notities
Carbon Design System gebruikt een overkoepelend UI Shell component. Hierin kun je gebruik maken van de Header Base with navigation, Actions, Side navigation (left panel).

**Comment by emma-koole** [comment]
## Helsinki Design System

[Storybook](https://hds.hel.fi/storybook/react/?path=/story/components-sidenavigation--default)

---

### Screenshots / Varianten
**Storybook**

Default
![Side navigation - default](https://github.com/user-attachments/assets/648d2558-cdd1-4d1d-89b9-de79b2326325)

With Icons
![Side navigation - with icons](https://github.com/user-attachments/assets/2a39938e-34f7-4b90-a468-fe569f0d094c)

Hover
![Side navigation - hover](https://github.com/user-attachments/assets/51f12bf4-5992-4160-b648-9d8f27b628b5)

**Comment by eslook** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Side Navigation op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=side+navigation)
